### PR TITLE
Support wireless debugging for iOS 12 or earlier

### PR DESCRIPTION
**Pull request by jaeyoi**
`idevicesyslog` requires the `--network` flag to obtain logs for iOS devices when wirelessly paired. 

When running Flutter on devices with iOS 12 or earlier versions, [the `idevicesyslog` command is used.](https://github.com/flutter/flutter/blob/5931b4f21da0c564b28794f40257b5f660596a3e/packages/flutter_tools/lib/src/ios/devices.dart#L1269-L1277).

Related Issue: #15072
Related PRs: #118104, #118895, #60623

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/ios/mac_test.dart [hunk 1]
```diff
@@ -44,6 +44,58 @@ void main() {
...
+        fakeProcessManager = FakeProcessManager.empty();
+      });
+
+      testWithoutContext('starts idevicesyslog when USB connected', () async {
+        fakeProcessManager.addCommand(const FakeCommand(
+          command: <String>[
+            'HostArtifact.idevicesyslog', '-u', '1234'
+          ],
+          environment: <String, String>{'DYLD_LIBRARY_PATH': '/path/to/libraries'},
+        ));
```
Comment (vashworth) [inline packages/flutter_tools/test/general.shard/ios/mac_test.dart:60]: The Flutter Style Guide now recommends [avoiding the use of `setUp`](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#make-each-test-entirely-self-contained).

Try this instead:

```suggestion
      testWithoutContext('starts idevicesyslog when USB connected', () async {
        final FakeProcessManager fakeProcessManager = FakeProcessManager.list(
          <FakeCommand>[
            const FakeCommand(
              command: <String>['HostArtifact.idevicesyslog', '-u', '1234'],
              environment: <String, String>{
                'DYLD_LIBRARY_PATH': '/path/to/libraries'
              },
            ),
          ],
        );
```

Comment (vashworth) [inline packages/flutter_tools/test/general.shard/ios/mac_test.dart:60]: The Flutter Style Guide now recommends [avoiding the use of `setUp`](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#make-each-test-entirely-self-contained).

Try this instead:

```suggestion
      testWithoutContext('starts idevicesyslog when USB connected', () async {
        final FakeProcessManager fakeProcessManager = FakeProcessManager.list(
          <FakeCommand>[
            const FakeCommand(
              command: <String>['HostArtifact.idevicesyslog', '-u', '1234'],
              environment: <String, String>{
                'DYLD_LIBRARY_PATH': '/path/to/libraries'
              },
            ),
          ],
        );
```

File: packages/flutter_tools/test/general.shard/ios/mac_test.dart [hunk 2]
```diff
@@ -44,6 +44,58 @@ void main() {
...
+        expect(fakeProcessManager, hasNoRemainingExpectations);
+      });
+
+      testWithoutContext('starts idevicesyslog when wirelessly connected', () async {
+        fakeProcessManager.addCommand(const FakeCommand(
+          command: <String>[
+            'HostArtifact.idevicesyslog', '-u', '1234', '--network'
+          ],
+          environment: <String, String>{'DYLD_LIBRARY_PATH': '/path/to/libraries'},
+        ));
```
Comment (vashworth) [inline packages/flutter_tools/test/general.shard/ios/mac_test.dart:82]: Same as comment above.

**Review by vashworth — CHANGES_REQUESTED** [review]
Thanks for contributing! Requested a small change to the test, but other than that, looks good!

Just curious, were you able to test with an actual iOS 12 device?

**Comment by jaeyoi** [comment]
@vashworth 
Thank you for the kind review. I have modified the code according to your comments.

> Just curious, were you able to test with an actual iOS 12 device?

I have successfully tested it using iPhone 5S (iOS 12.5.7).

**Review by vashworth — APPROVED** [review]
Thanks for the changes!

**Review by jmagman — APPROVED** [review]
LGTM, thanks for the contribution!

**Pull request merged**